python_modules/dagster-ext/dagster_ext/_io/default.py

```diff
@@ -45,6 +67,14 @@ def write_message(self, message: ExtMessage) -> None:
             f.write(json.dumps(message) + "\n")
 
 
+class ExtStreamMessageChannel(ExtMessageWriterChannel):
+    def __init__(self, stream: TextIO):
+        self._stream = stream
+
+    def write_message(self, message: ExtMessage) -> None:
+        self._stream.writelines((json.dumps(message), "\n"))
```

Comment:
testing this in the existing suite would involve modifications to  `ExtSubprocess` in the same vein as highlighted here in k8s https://github.com/dagster-io/dagster/pull/16122/files#r1313558644

The crux of the problem here is needing to interact with post launch information in order to do reads.